### PR TITLE
Milestone 10. Add authorization rules

### PR DESCRIPTION
**Pull request by indigodavid**
# Milestone 10. Add authorization rules 🎉 

For this milestone, the following features have been implemented:

## User Authorization Feature

- Added gem `cancancan` -> Library to handle user authorizations
- Add role column to Users table
- Add method admin? to User model to check for 'admin' role

## New Model Ability

- CanCan Generated
- Provides permissions for authors to delete their comments and posts
- Allows 'admin' role to manage all the objects

## View fixes

- Post -> Adds a delete button to remove Post if user is authorized
- Comment -> Adds a delete button to remove Comment if user is authorized 

## Model fixes

- Like -> Added after_destroy method to update its counter
- Comment -> Added after_destroy method to update its counter
- Post -> Added after_destroy method to update its counter
- Fixed dependencies when destroying objects

## Controller fixes

Added destroy methods for `Like`, `Comment`, and `Post`

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -27,6 +27,13 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    @post.destroy
+
+    redirect_to user_path(params[:user_id]), status: :see_other
+  end
+
```
Comment (Lordkaito) [inline app/controllers/posts_controller.rb:38]: - [x] Please make sure to add the authorization here, on your posts controller. You should add `load_and_authorize_resource` at the top of the controller in order to make sure that `CanCanCan` gem works properly.

**Review by Lordkaito — CHANGES_REQUESTED** [review]
Hi @indigodavid, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (indigodavid) [reply app/controllers/posts_controller.rb:38]: @Lordkaito I did the change as requested. Thank you for improving our App

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED ✔️ 🟢 

## Hi team 👋, 

Well done for implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by indigodavid** [comment]
@codepantha Thank you for your kind review.

I'll merge my branch right away!! 😃 💻

**Pull request merged**